test/src/e2e_vm_tests/test_programs/should_pass/resolve_local_items_that_shadow_imports/test.toml

```diff
@@ -0,0 +1,23 @@
+category = "compile"
+validate_abi = true
+expected_warnings = 6
+
+#check: $()warning
+#check: $()pub enum Enum {
+#nextln: $()This enum is never used.
+
+#check: $()warning
+#check: $()pub struct Struct {
+#nextln: $()This struct is never used.
+
+#check: $()warning
+#check: $()pub struct PubStruct {
+#nextln $()This struct is never used.
+
+#check: $()warning
+#check: $()pub struct GenericStruct<T> {
+#nextln: $()This struct is never used.
+
+#check: $()warning
+#check: $()pub const X: u64 = 0u64;
+#nextln: $()This declaration is never used.
```

Comment:
Are these useful warnings? Shouldn't it be fine to import all declarations from a library (using `use lib::*;`) and then not use all of them?

In this example none of the imported declarations are used, but then the warning should be something like "Unused import 'use lib::*;'".

Reply:
This is a great question, but out of scope of this PR :smile: It's a question for our DCA and how it works with local libraries, those used in a project that is not only a library. To my understanding, those items are not expected to be reused outside of the project and this if they are public and not used anywhere in the project they trigger dead code warnings. I _think_ it works that way.

BDW I was thinking about the same things while working on making struct fields private by default.

Anyhow, my proposal is to open a separate issue to discuss this topic or get clarifications.

For now, since these warnings are there I am (mis?)using them to additionally prove that those types are really not resolved in the module.